### PR TITLE
test: refactor, add setup.py, add tests

### PR DESCRIPTION
**Pull request by tbpg**
I updated the README with instructions for testing and development.

I used nox to follow the pattern of other Python repos. That led to
needing to add a setup.py, since nox requires it.

I used TEST_BUCKET instead of SOURCE_BUCKET to make it
clear it's for testing.

I refactored the generation logic into a generate.py file to make it
easier to test.

setup.py includes dependencies, so we can remove requirements.txt.

### CODE REVIEW

File: ci/common.cfg [hunk 1]
```diff
@@ -42,3 +42,8 @@ env_vars: {
     key: "TRAMPOLINE_DOCKERFILE"
     value: "docfx/Dockerfile"
 }
+
+env_var: {
```
Comment (tmatsuo) [inline ci/common.cfg:46]: I think it must be `env_vars`

File: tests/test_generate.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+def yaml_dir(tmpdir):
+    shutil.copytree("testdata", tmpdir, dirs_exist_ok=True)
+    return tmpdir
+
+
+def test_generate(yaml_dir, tmpdir):
+    test_bucket = os.environ.get("TEST_BUCKET")
+    if not test_bucket:
+        pytest.skip("Must set TEST_BUCKET")
+    return test_bucket
```
Comment (tmatsuo) [inline tests/test_generate.py:36]: I think `return` here will end the test.

**Comment by tmatsuo** [comment]
I think that box makes sense if you need to support multiple versions of Python (client libs and samples), but for this repo, I'm pretty happy if it works in our Docker image.

Can we not use nox and simplify things?

File: ci/common.cfg (hunk 1 above)
Comment (tbpg) [reply ci/common.cfg]: Done.

File: tests/test_generate.py (hunk 2 above)
Comment (tbpg) [reply tests/test_generate.py]: Whoops! Fixed. Good catch.

**Pull request merged**